girder/web_client/test/spec/browserSpec.js

```diff
@@ -974,3 +977,275 @@ describe('browser hierarchy selection', function () {
+                titleText: 'This is a title',
+                defaultSelectedResource: itemlist[itemlist.length - 1],
+                selectItem: true,
+                paginated: true,
+                highlightItem: true,
+                showItems: true
+            }).render();
+        });
+
+        waitsFor(function () {
+            return $(widget.$el).is(':visible');
+        });
+
+        waitsFor(function () {
+            return $('.g-hierarchy-widget').length > 0 &&
+                               $('.g-item-list-link').length > 0;
+        }, 'the hierarchy widget to display');
+
+        waitsFor(function () {
+            return $('#g-page-selection-input').val() === '2';
+        }, 'waits for it to go to the second page');
+
+        runs(function () {
+            expect(widget.$('#g-selected-model').val()).toBe('item#: 99');
+            expect($('.g-hierarachy-paginated-bar').length).toBe(1);
+            expect($('#g-page-selection-input').val()).toBe('2');
+            expect($('.g-hierarchy-breadcrumb-bar').hasClass('g-hierarchy-sticky')).toBe(true);
+        }, 'Make sure paginated text is displayed with proper settings');
+
+        runs(function () {
+            $('.g-folder-list-link').trigger('click');
+        });
+
+        waitsFor(function () {
+            return $('.g-hierarachy-paginated-bar').length === 0;
+        }, 'The removal of the page selection');
+
+        runs(function () {
+            expect($('.g-hierarachy-paginated-bar').length).toBe(0);
+            $('.g-breadcrumb-link[g-index="1"]').trigger('click');
+        }, 'Make sure paginated text is removed when a folder has less than one page');
+
+        waitsFor(function () {
+            return $('.g-hierarachy-paginated-bar').length === 1 && $('.g-item-list-entry').length > 10;
+        }, 'The removal of the page selection');
+
+        waitsFor(function () {
+            return $('#g-page-selection-input').val() === '1';
+        }, 'waits for it to go to the second page');
+
+        runs(function () {
+            expect($('.g-hierarachy-paginated-bar').length).toBe(1);
+            expect($('#g-page-selection-input').val()).toBe('1');
+            expect($('.g-hierarchy-breadcrumb-bar').hasClass('g-hierarchy-sticky')).toBe(true);
+        }, 'Returning to the folder should bring us to the first page');
+
+        runs(function () {
+            $('#g-page-selection-input').val(2);
+            $('#g-page-selection-input').trigger('change');
+        }, 'Change the page by using the input field');
+
+        waitsFor(function () {
+            return $('#g-page-selection-input').val() === '2';
+        });
+
+        runs(function () {
+            expect($('.g-hierarachy-paginated-bar').length).toBe(1);
+            expect($('#g-page-selection-input').val()).toBe('2');
+            expect($('.g-hierarchy-breadcrumb-bar').hasClass('g-hierarchy-sticky')).toBe(true);
+        }, 'Should be on a second page now');
```

Comment:
Perhaps now add following the previous and the next buttons (I don't think they are exercised in the tests currently).